tests/tracing/conftest.py

```diff
@@ -92,3 +95,39 @@ def databricks_tracking_uri():
 def async_logging_enabled(request, monkeypatch):
     monkeypatch.setenv(MLFLOW_ENABLE_ASYNC_LOGGING.name, str(request.param))
     return request.param
+
+
+@pytest.fixture
+def otel_collector():
+    """Start an OpenTelemetry collector in a Docker container."""
+    subprocess.run(["docker", "pull", "otel/opentelemetry-collector-contrib"], check=True)
+
+    output_file = tempfile.NamedTemporaryFile().name
```

Comment:
```suggestion
    with tempfile.NamedTemporaryFile() as tmp_file:
```

can we use a context manager?